File: January2022/Python/canPlaceFlowers.py
# Author :  Karthik K (https://www.linkedin.com/in/karthikmunirathinam/)

# TC : O(N)
# SC : O(1)

class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        ind = 0
        while ind < len(flowerbed):
            if flowerbed[ind] == 0 and (ind == 0 or flowerbed[ind-1] == 0) and (ind == len(flowerbed)-1 or flowerbed[ind+1] == 0):
                n -= 1
                flowerbed[ind] = 1
                ind += 1
            if n <= 0:
                return True
            ind += 1
        return False

File: January2022/Java/CanPlaceFlowers.java
public class Solution {
	// TC: O(n)
	// SC : O(1)
	public boolean canPlaceFlowers(int[] flowerbed, int n) {

		for(int i = 0; i < flowerbed.length && n>0;  i++) {
			if(flowerbed[i] == 0) {
				int next = (i == flowerbed.length - 1) ? 0 : flowerbed[i + 1];
				int prev = (i == 0) ? 0 : flowerbed[i - 1];
				if(next == 0 && prev == 0) {
					flowerbed[i] = 1;
					n--;
				}
			}
		}

		return  n==0;
	}
}